pkg/app/piped/sourceprocesser/attacher.go

```diff
@@ -25,53 +25,61 @@ import (
 	"github.com/pipe-cd/pipecd/pkg/config"
 )
 
-func AttachData(appDir string, atc config.Attachment) error {
-	if len(atc.Targets) == 0 {
-		return nil
+type attachmentProcessor struct {
+	atc *config.Attachment
+}
+
+func NewAttachmentProcessor(atc *config.Attachment) *attachmentProcessor {
+	return &attachmentProcessor{
+		atc: atc,
 	}
-	if len(atc.Sources) == 0 {
-		return fmt.Errorf("no source data to attach")
+}
+
+func (p *attachmentProcessor) BuildTemplateData(appDir string) (map[string]string, error) {
+	if len(p.atc.Sources) == 0 {
+		// Skip building no error.
+		return nil, nil
 	}
 
-	src := make(map[string]string, len(atc.Sources))
-	for k, v := range atc.Sources {
+	src := make(map[string]string, len(p.atc.Sources))
+	for k, v := range p.atc.Sources {
 		srcPath := filepath.Join(appDir, v)
 		buff, err := os.ReadFile(srcPath)
 		if err != nil {
-			return fmt.Errorf("failed to read data source to attach from file %s (%w)", v, err)
+			return nil, fmt.Errorf("failed to read data source to attach from file %s (%w)", v, err)
 		}
 		src[k] = string(buff)
 	}
-	data := map[string](map[string]string){
-		"attachment": src,
-	}
+	return src, nil
+}
+
+func (p *attachmentProcessor) TemplateKey() string {
+	return "attachment"
```

Comment:
[IMO] define it as const value 👀

Reply:
this object function acts as constant for this "object" already, so I don't think we need to make new constant 🌝